compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala

```diff
@@ -441,9 +449,8 @@ case class ApplicationCompiler(typeAliases: Map[String, Type],
         // The reorg applet requires higher permissions to organize the output directory.
         Some(DxAccess.empty.copy(project = Some(DxAccessLevel.Contribute)))
       case _ =>
-        // Scatters need network access, because they spawn subjobs that (may) use dx-docker.
-        // We end up allowing all applets to use the network
-        Some(DxAccess.empty.copy(network = Some(Vector("*"))))
```

Comment:
@commandlinegirl here we can just make a new case to give `ExecutableKindWfFragment` full access and restrict the other kinds  `ExecutableKindWfInputs / ExecutableKindWfOutputs / ExecutableKindWfCustomReorgOutputs`